readiness-tool/popup.js

```diff
@@ -0,0 +1,231 @@
+  notSupportedOther.appendChild(makeList(detectedApps.notSupported.other));
+}
+
+/**
+ * Splits all detected apps into 'supported' and 'not supported'
+ * @param {Object} apps - All possible services that could exist on the page
+ * @param {String} htmlString - String containing all HTML on the page
+ * @return {Object} 
+ */
+function specifySupport(apps, htmlString) {
+  let obj = apps;
+  let foundThis = {
+    "supported": {
+      "ads": [],
+      "analytics": [],
+      "other": []
+    },
+    "notSupported": {
+      "ads": [],
+      "analytics": [],
+      "other": []
+    }
+  };
+
+  //for all the app objects in the apps.JSON file
+  for (let key in obj) {
+    if (obj.hasOwnProperty(key)) {
+      let val = obj[key];
+      //If object has a 'script' key
+      if (val.script != null) {
+        // Sometimes val.script contains an array of regular expressions to check
+        if (typeof (val.script) == 'object') {
+          for (let x in val.script) {
+            let tempScript = val.script[x].split('\\;');
+            addToDict(tempScript, htmlString, foundThis, key, val.cats);
+          }
+        } else {
+          let tempScript = val.script.split('\\;');
+          addToDict(tempScript, htmlString, foundThis, key, val.cats);
+        }
+
+      }
+    }
+  }
+  return foundThis;
+}
+
+
+/**
+ * Pushes keys to either the supported or not supported list of the object 'found this'
+ * @param {Array} tempScript - String array containing the regular expression being looked for
+ * @param {!String} htmlString - String containing all HTML on the page
+ * @param {Object} foundThis - Object seperating the 3p services based on available support
+ * @param {String} key - name of third party service
+ * @param {String} category - the category that the key belongs to
+ */
+function addToDict(tempScript, htmlString, foundThis, key, category) {
+  let regX = new RegExp(tempScript[0]);
+  if (doesRegexExist(regX, htmlString)) {
+    if (get(foundThis, key) == undefined) {
+      if (isSupported(key)) {
+        findCategory(category, foundThis.supported, key);
+      } else {
+        findCategory(category, foundThis.notSupported, key);
+      }
+    }
+  }
+}
+
+/**
+ * Checks to see if this key is in the object already
+ * @param {Object} obj - Object seperating the 3p services based on available support
+ * @param {String} key - name of third party service
+ * @return {boolean} 
+ */
+function get(obj, key) {
+  return key.split(".").reduce(function (o, x) {
+    return (typeof o == "undefined" || o === null) ? o : o[x];
+  }, obj);
+}
+
+/**
+ * CFigure out which category the key belongs to (either - Ads, Analytics or Other)
+ * @param {String} category - number representing the category that each item belongs to
+ * @param {Object} objectList - The found supported/unsupported 3p services
+ * @param {String} key - name of third party service
+ */
+function findCategory(category, objectList, key) {
+  if (category == "10") {
```

Comment:
add comments that describe what "10" and "36" mean.
Also - change the double quotes to single quotes: '10', '36'